taskvine/src/bindings/python3/ndcctools/taskvine/dask_executor.py

```diff
@@ -272,10 +272,14 @@ def _dask_execute(self, dsk, keys):
                         if self.wrapper:
                             self.wrapper_proc(t.load_wrapper_output(self))
 
-                        result_file.garbage_collect_children(self)
-
                         if t.key in dsk:
                             bar_update(advance=1)
+
+                        children = dag.get_children(t.key)
```

Comment:
Move this code to a method, like `DaskVine.prune_cluster` or something like that.